sql/src/main/java/org/apache/druid/sql/AbstractStatement.java

```diff
@@ -160,7 +160,12 @@ protected void authorize(
 
     // Authentication is done by the planner using the function provided
     // here. The planner ensures that this step is done before planning.
-    authResult = planner.authorize(authorizer, contextResources);
+    try {
+      authResult = planner.authorize(authorizer, contextResources);
+    } catch (Exception e) {
+      log.error(e, "Planner threw an exception during authorization");
+      throw new RuntimeException("Could not authorize resources for query", e);
+    }
```

Comment:
Why do we need this?

This code translates the unauthorized exception into a runtime exception, which will change the code flow. The desired flow is that we throw the unauthorized exception which is caught by a Jersey handler up the hander stack. (I seem to recall issues when I tried to intercept that error as you've done here.)

If we want to log auth errors, that would be the place. Else, in the SQL resource, we already log errors. And, the statement logs failures at the end of the query process.

As a result, this step seems unnecessary. What problem is this trying to solve?